pandas/core/frame.py

```diff
@@ -7994,6 +8018,47 @@ def blk_func(values):
             result = self._constructor_sliced(result, index=labels)
         return result
 
+    def _reduce_columns(self, op, name, skipna=True, **kwds):
+        """
+        Reduce DataFrame column-wise.
+
+        Parameters
+        ----------
+        op : func
+            The reducing function to be called on the values. Only used
+            for columns backed by a numpy ndarray.
+        name : str
+            The name of the reduction.
+        skipna, **kwds : keywords to pass to the `op` function
+
+        Returns
+        -------
+        Series
+        """
+        result = []
+
+        for arr in self._iter_arrays():
+            if isinstance(arr, ExtensionArray):
+                # dispatch to ExtensionArray interface
+                val = arr._reduce(name, skipna=skipna, **kwds)
+            else:
+                # dispatch to numpy arrays
+                with np.errstate(all="ignore"):
+                    val = op(arr, skipna=skipna, **kwds)
+
+            result.append(val)
+
+        return self._constructor_sliced(result, index=self.columns)
+
+    def _iter_arrays(self):
```

Comment:
if this is only used once, id rather inline it.  if we do add this method, it should probably go adjacent to [similar methods, not sure off the top of my head]

Reply:
Yeah, the location where to put it needs to be decided (if we keep it), just put it here because I was editing the code here.

I think it will be useful for more things in the future (anything that we want to optimize for column-wise), but we can also extract it at that moment that we would need it

Reply:
Inlined for now